pwiz_tools/Skyline/SettingsUI/Irt/CalibrateIrtDlg.cs

```diff
@@ -692,46 +693,67 @@ private static List<MeasuredPeptide> FindBestPeptides(SrmDocument doc, int pepti
             /// 
             /// The returned list is guaranteed to be sorted by retention time.
             /// </summary>
+            /// <param name="doc">Document containing the peptides to choose from</param>
             /// <param name="docPeptides">Peptides to choose from</param>
             /// <param name="peptideCount">The number of peptides desired</param>
-            private static List<MeasuredPeptide> FindEvenlySpacedPeptides(List<MeasuredPeptide> docPeptides, int peptideCount)
+            private static List<MeasuredPeptide> FindEvenlySpacedPeptides(SrmDocument doc, List<MeasuredPeptide> docPeptides, int peptideCount)
             {
-                docPeptides.Sort((x, y) => x.RetentionTime.CompareTo(y.RetentionTime));
                 if (docPeptides.Count == peptideCount)
+                {
+                    docPeptides.Sort((x, y) => x.RetentionTime.CompareTo(y.RetentionTime));
                     return docPeptides;
+                }
+
+                var model = doc.Settings.PeptideSettings.Integration.PeakScoringModel;
+                if (model == null || !model.IsTrained)
+                    model = LegacyScoringModel.DEFAULT_MODEL;
 
-                var minRT = docPeptides.First().RetentionTime;
-                var maxRT = docPeptides.Last().RetentionTime;
-
-                /*
-                 * This algorithm will pick the closest peptide to each "target RT" as defined
-                 * by the "length of the gradient" (Last peptide's RT - First peptide's RT) and
-                 * the number of peptides asked for.
-                 * 
-                 * It does this by considering peptides 3 at a time: (prev, current, next) triplets.
-                 * When the pointer has shifted so that the "current" peptide is closer than
-                 * either of its neighbors, then that peptide is added to the standard and removed
-                 * from the search list.
-                 */
-                List<MeasuredPeptide> standardPeptides = new List<MeasuredPeptide>();
-                double gradientLength = maxRT - minRT;
-                for (int i = 0; i < peptideCount; i++)
-                {
-                    double targetRT = minRT + i * (gradientLength / (peptideCount - 1));
-                    for (int j = 0; j < docPeptides.Count; j++)
+                var resultsHandler = new MProphetResultsHandler(doc, model);
+                resultsHandler.ScoreFeatures(null, true);
+
+                var docTargets = docPeptides.ToDictionary(pep => pep.Target, pep => pep);

```

Comment:
I think you need to guard against an exception caused by the duplicate peptides here.